### DISCUSSION TITLE
Dbeaver - Git Solution

### DISCUSSION THREAD

**Discussion by priyank04** [dbeaver/dbeaver#21530] [Q&A] [answered]
Hey Community,

Happy Fall! 🍁

I am looking for ways to leverage dbeaver git extension and implement a continuous process of scripting out the DDL queries of tables and version controlling them using GitHub. I am generating SQL once for each table and storing them in Scripts folder. For any change in structure, I am regenerating SQL and storing in the script created earlier. Now, I would like to understand how can I use dbeaver `git` or `team synchronize` perspective to:

- use fine-grained tokens to setup git repo
- maintain branches (checkout, compare, merge)
- create Pull Requests (main branch is locked)

Any documentation/advises towards this would be really helpful.

**Comment by E1izabeth** [comment]
Here is a wiki about git integration https://github.com/dbeaver/dbeaver/wiki/Project-team-work

**Reply by priyank04** [reply]
That is really helpful. But one thing is missing on, how to work with a repository using Fine-grained token, cloning URI and pushing? Any guide on that would be helpful.

**Reply by priyank04** [reply]
@E1izabeth, Reaching out back to check, do you have any more documentation on using GitHub fine grained tokens with DBeaver?

**Reply by E1izabeth** [reply]
We don't have a documentation about that

**Reply by priyank04** [reply]
Does your team plan to add some documentation anytime soon on it? 

Given Dbeaver captures the Git URI when cloning the repository, so when using the fine-grained token, secret token is to be passed in URI. So next time onwards when a pull/push needs to be made, Tte URI is uneditable and do not have the tokein in the uri and the pull/push never goes through. 

I want to limit repositories which can be accessed, which is possible only using fine grained tokens as per my understanding. I am new to the platform hence more documentation on it would really be helpful.

**Reply by flambchap** [reply, marked as answer]
Hi @priyank04. I haven't tested Git integration with these new tokens, but I believe they should work similarly to classic tokens, meaning you need to pass your token to the password field.

**Reply by priyank04** [reply]
Fine-grained tokens didn't work for me the way it worked with classic tokens. 
I had an issue to connect a repository which was solved earlier [through community here](https://github.com/dbeaver/dbeaver/issues/20992) where I used classic tokens.

Tried the same with fine-grained tokens but didn't work that way. I used [this solution](https://stackoverflow.com/questions/42148841/github-clone-with-oauth-access-token) and it worked for me. But when making push or pull requests my URI would just have oath2@github.com or similar without token.

**Reply by flambchap** [reply]
>Tried the same with fine-grained tokens but didn't work that way

I've just tried it, and it works fine for me. What issues have you encountered? What permissions have you granted for the token?

**Reply by priyank04** [reply]
I overviewed the permissions. And found out the permissions were not enough for the repository to make commits. Also it works as the password too. Thanks for quick help over the matter folks. 

**Fine-grained tokes** can be used same way as Classic ones.
